### DISCUSSION TITLE
Can prawn create "linearized" PDFs?

### DISCUSSION THREAD

**Discussion by jrochkind** [prawnpdf/prawn#1298] [General]
see:

> A linearized PDF file is a special format of a PDF file that makes viewing faster over the Internet. Linearized PDF files contains information that allow a byte-streaming server to download the PDF file one page at a time. If byte-streaming is disabled on the server or if the PDF file is not linearized, the entire PDF file must be downloaded before it can be viewed. 

https://docs.oracle.com/cd/E96927_01/TSG/FAQ/What%20are%20linearized%20PDF%20files_.html

and

https://blog.idrsolutions.com/what-is-a-linearized-pdf/

**Comment by pointlessone** [comment]
No, Prawn doesn't support linearization as of 2.4.

I don't think there's an easy workaround as Linearization requires multiple xref tables which Prawn doesn't support either.
